sklearn/linear_model/_base.py

```diff
@@ -361,8 +376,6 @@ def predict(self, X):
         """
         return self._decision_function(X)
 
-    _preprocess_data = staticmethod(_preprocess_data)
```

Comment:
Could it be a method ? I'm ok with it being a method or a function, with a slight preference for a method if it's only meant to be used within the class.

Reply:
I'm trying to disentangle the whole `_preprocess_data` and `_pre_fit` in linear models. This static method is nonsense. We can always just directly call the normal function `_preprocess_data`.

Reply:
The static method is not necessary, but it could be a normal method. Is the function used outside of LinearModel ?

Anyway as I said I'm also fine with it being a function

Reply:
The function `_preprocess_data` (line 213) is only used inside `linear_model`.